test/configuration_test.dart

```diff
@@ -153,4 +153,186 @@ Future<void> main([List<String>? args]) async {
     realm1.close();
     realm2.close();
   });
+
+  test('Configuration.initialDataCallback invoked', () {
+    var invoked = false;
+    var config = Configuration([Dog.schema, Person.schema]);
+    config.initialDataCallback = (realm) {
+      invoked = true;
+      realm.add(Dog('fido', owner: Person('john')));
+    };
+
+    final realm = Realm(config);
+    expect(invoked, true);
+
+    final people = realm.all<Person>();
+    expect(people.length, 1);
+    expect(people[0].name, 'john');
+
+    final dogs = realm.all<Dog>();
+    expect(dogs.length, 1);
+    expect(dogs[0].name, 'fido');
+
+    realm.close();
+  });
+
+  test('Configuration.initialDataCallback can be unset', () {
+    var invoked = false;
+    var config = Configuration([Dog.schema, Person.schema]);
+    config.initialDataCallback = (realm) {
+      invoked = true;
+      realm.add(Dog('fido', owner: Person('john')));
+    };
+    config.initialDataCallback = null;
```

Comment:
This obviously won't be possible with my idea, but instead you can just choose not to await the `initialData` future which will have the same effect.